src/fpu/fpu.cpp

```diff
@@ -27,39 +27,51 @@
 #include "fpu.h"
 #include "cpu.h"
 
-FPU_rec fpu;
+FPU_rec fpu = {};
 
 void FPU_FLDCW(PhysPt addr){
 	uint16_t temp = mem_readw(addr);
 	FPU_SetCW(temp);
 }
 
-uint16_t FPU_GetTag(void){
-	uint16_t tag=0;
-	for(Bitu i=0;i<8;i++)
-		tag |= ( (fpu.tags[i]&3) <<(2*i));
+uint16_t FPU_GetTag()
+{
+	uint16_t tag = 0;
+	for (Bitu i = 0; i < 8; i++) {
+		tag |= ((fpu.tags[i] & 3) << (2 * i));
+	}
 	return tag;
 }
 
-void FPU_LoadPRegs(const uint8_t* buffer)
+void FPU_SetPRegsFrom(const uint8_t dyn_regs[8][10])
```

Comment:
I think the original method of copying from a typeless buffer is okay.  This is essentially copying from a buffer saved by the FSAVE machine instruction.  There is no point to attempt to give it some structure.  The compiler can optimize fixed size memcpy very effectively (https://godbolt.org/z/Pv3fch5fM)

Reply:
> I think the original method of copying from a typeless buffer is okay. 

No argument; it gets the job done.

Unfortunately only partially writing structs and class data (via bulk memcpy and memset) often causes compilers to caution that we might be leaving stuff behind. 

![2023-01-27_13-17](https://user-images.githubusercontent.com/1557255/215202518-c641dee5-111d-463a-ae44-238397562f2a.png)
![2023-01-27_13-16](https://user-images.githubusercontent.com/1557255/215202522-e6fbb864-df94-41cd-a8e0-3c175f74e2fe.png)


It's a good warning because when someone changes a struct definition, the impact can ripple to all the memcpy/memset's that are using hard-coded quantities.

The classic solution is using sizeof on the struct to set memcpy's quantity to keep it in lock-step with the definition. 

And no doubt, you would have done that here too, but I realize we're only after a deliberate subset of the struct members.

Under the hood, the `unaligned_` accessors also use memcpy to access the bytes and cast up the result to bonafide types: and like you, confirmed that those memcpy's get translated to essentially the same code as the memory type-casters (so zero performance impact).

In this case: I realize these aren't available on other DOSBoxes: so I suggest the patches still use the memcpy (and those projects can assess those warnings).

Of course, they are also welcome to grab Staging's unaligned accessors. They are tiny and extensively tested.

```text
include/mem_unaligned.h:static inline uint16_t read_unaligned_uint16(const uint8_t *arr) noexcept
include/mem_unaligned.h:static inline uint32_t read_unaligned_uint32(const uint8_t *arr) noexcept
include/mem_unaligned.h:static inline uint64_t read_unaligned_uint64(const uint8_t *arr) noexcept
include/mem_unaligned.h:static inline size_t read_unaligned_size_t(const uint8_t *arr) noexcept
include/mem_unaligned.h:static inline uint16_t read_unaligned_uint16_at(const uint8_t *arr,
include/mem_unaligned.h:static inline uint32_t read_unaligned_uint32_at(const uint8_t *arr,
include/mem_unaligned.h:static inline uint64_t read_unaligned_uint64_at(const uint8_t *arr,
include/mem_unaligned.h:static inline size_t read_unaligned_size_t_at(const uint8_t *arr,
include/mem_unaligned.h:static inline void write_unaligned_uint16(uint8_t *arr, const uint16_t val) noexcept
include/mem_unaligned.h:static inline void write_unaligned_uint32(uint8_t *arr, const uint32_t val) noexcept
include/mem_unaligned.h:static inline void write_unaligned_uint64(uint8_t *arr, const uint64_t val) noexcept
include/mem_unaligned.h:static inline void write_unaligned_uint16_at(uint8_t *arr,
include/mem_unaligned.h:static inline void write_unaligned_uint32_at(uint8_t *arr,
include/mem_unaligned.h:static inline void write_unaligned_uint64_at(uint8_t *arr,
include/mem_unaligned.h:static inline void add_to_unaligned_uint16(uint8_t *arr, const uint16_t val) noexcept
include/mem_unaligned.h:static inline void add_to_unaligned_uint32(uint8_t *arr, const uint32_t val) noexcept
include/mem_unaligned.h:static inline void add_to_unaligned_uint64(uint8_t *arr, const uint64_t val) noexcept
include/mem_unaligned.h:static inline void inc_unaligned_uint16(uint8_t *arr) noexcept
include/mem_unaligned.h:static inline void inc_unaligned_uint32(uint8_t *arr) noexcept
include/mem_unaligned.h:static inline void inc_unaligned_uint64(uint8_t *arr) noexcept
```

Reply:
That is  one `memcpy` vs  3 `memcpy` (and maybe some extra assignments)